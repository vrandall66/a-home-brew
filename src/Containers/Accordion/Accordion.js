import React, { Component } from 'react';
import dompurify from 'dompurify';
import './Accordion.scss';

export class Accordion extends Component {
  componentDidMount = () => {
    const clean = dompurify.sanitize;
    console.log(content);
  };
  render() {
    return (
      <div className='Accordion__section'>
        <button className='Accordion'>
          <p className='Accordion__title'>{title}</p>
        </button>
        <div className='Accordion__content'>
          <div
            className='Accordion__text'
            dangerouslySetInnerHTML={{
              __html: clean({ content })
            }}
          />
        </div>
      </div>
    );
  }
}

export default Accordion;
