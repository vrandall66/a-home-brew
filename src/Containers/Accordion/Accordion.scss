* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.Accordion__section {
  grid-area: nav;
}

.Accordion {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  display: flex;
  align-items: center;
  border: none;
  outline: none;
  transition: background-color 0.6s ease;
}

.Accordion:hover,
.active {
  background-color: #ccc;
}

.Accordion__title {
  font-family: 'Open Sans', sans-serif;
  font-weight: 600;
  font-size: 14px;
}

.Accordion__icon {
  margin-left: auto;
  transition: transform 0.6s ease;
}

.rotate {
  transform: rotate(90deg);
}

.Accordion__content {
  background-color: white;
  overflow: hidden;
  transition: max-height 0.6s ease;
}

.Accordion__text {
  font-family: 'Open Sans', sans-serif;
  font-weight: 400;
  font-size: 14px;
  padding: 18px;
}
