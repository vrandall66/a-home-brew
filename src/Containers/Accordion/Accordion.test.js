import React from 'react';
import { shallow } from 'enzyme';
import Accordion from './Accordion';

describe('Accordion', () => {
  let wrapper;
  let hop = {
    title: 'Fuggles',
    amount: {
      value: 25,
      unit: 'grams'
    },
    add: 'start',
    attribute: 'bitter',
    listItems: [`Attribute: Fuggles`, `When to add it: start`, `25 grams`]
  };
  let index = 1

  beforeEach(() => {
    wrapper = shallow(
      <Accordion title={hop.title} items={hop.listItems} key={index} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});
