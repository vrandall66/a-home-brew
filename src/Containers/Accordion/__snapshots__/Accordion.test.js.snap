// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should match the snapshot 1`] = `
<div
  className="Accordion__section"
>
  <button
    className="Accordion"
    onClick={[Function]}
  >
    <p
      className="Accordion__title"
    >
      Fuggles
    </p>
    <IoIosArrowDropdown
      className="Accordion__icon"
    />
  </button>
</div>
`;
